src/CFamily/CompilationDatabase/CompilationDatabaseLocator.cs

```diff
@@ -0,0 +1,130 @@
+    {
+        internal const string CompilationDatabaseFileName = "compile_commands.json";
+        internal const string CMakeSettingsFileName = "CMakeSettings.json";
+        internal const string DefaultConfiguration = "x64-Debug";
+        internal const string DefaultLocationFormat = "{0}\\out\\build\\{1}";
+
+        private readonly IFolderWorkspaceService folderWorkspaceService;
+        private readonly IFileSystem fileSystem;
+        private readonly ILogger logger;
+
+        [ImportingConstructor]
+        public CompilationDatabaseLocator(IFolderWorkspaceService folderWorkspaceService, ILogger logger)
+            : this(folderWorkspaceService, new FileSystem(), logger)
+        {
+        }
+
+        public CompilationDatabaseLocator(IFolderWorkspaceService folderWorkspaceService, IFileSystem fileSystem, ILogger logger)
+        {
+            this.folderWorkspaceService = folderWorkspaceService;
+            this.fileSystem = fileSystem;
+            this.logger = logger;
+        }
+
+        public string Locate()
+        {
+            var databaseFolderPath = GetCompilationDatabaseFolderPath();
+
+            if (string.IsNullOrEmpty(databaseFolderPath))
+            {
+                return null;
+            }
+
+            var compilationDatabaseFilePath = Path.Combine(databaseFolderPath, CompilationDatabaseFileName);
+
+            if (!fileSystem.File.Exists(compilationDatabaseFilePath))
+            {
+                logger.WriteLine(Resources.NoCompilationDatabaseFile, compilationDatabaseFilePath);
+
+                return null;
+            }
+
+            logger.WriteLine(Resources.FoundCompilationDatabaseFile, compilationDatabaseFilePath);
+
+            return compilationDatabaseFilePath;
+        }
+
+        public string GetCompilationDatabaseFolderPath()
+        {
+            var rootDirectory = folderWorkspaceService.FindRootDirectory();
+
+            if (string.IsNullOrEmpty(rootDirectory))
+            {
+                logger.WriteLine(Resources.NoRootDirectory);
+                return null;
+            }
+
+            var cmakeSettingsFullPath = Path.GetFullPath(Path.Combine(rootDirectory, CMakeSettingsFileName));
+            var activeConfiguration = DefaultConfiguration;
+
+            if (!fileSystem.File.Exists(cmakeSettingsFullPath))
+            {
+                var defaultLocation = Path.GetFullPath(string.Format(DefaultLocationFormat, rootDirectory, activeConfiguration));
+
+                logger.WriteLine(Resources.NoCMakeSettings, CMakeSettingsFileName, rootDirectory, defaultLocation);
+
+                return defaultLocation;
+            }
+
+            logger.WriteLine(Resources.ReadingCMakeSettings, cmakeSettingsFullPath);
+
+            var settingsString = fileSystem.File.ReadAllText(cmakeSettingsFullPath);
+            var settings = JsonConvert.DeserializeObject<CMakeSettings>(settingsString);
```

Comment:
This will throw if the json is invalid, won't it?

It's not clear how we want to handle this yet (i.e. catch here or higher up). It's worth adding a note to the board to test invalid json to make sure we don't crash VS.